modules/tinymce/src/core/test/ts/browser/EditorOptionsDebugTest.ts

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable no-console */
+import { context, describe, it } from '@ephox/bedrock-client';
+import { Fun, Arr, Obj, Type } from '@ephox/katamari';
+import { SugarElement } from '@ephox/sugar';
+import { TinyHooks } from '@ephox/wrap-mcagar';
+import { assert } from 'chai';
+
+import Editor from 'tinymce/core/api/Editor';
+
+describe('browser.tinymce.core.EditorOptionsDebugTest', () => {
+  const createHook = (options: Record<string, unknown>) =>
+    TinyHooks.bddSetupLight<Editor>({
+      ...options,
+      base_url: '/project/tinymce/js/tinymce',
+    }, []);
+  const ignoredKeys = new Set([ 'promotion', 'toolbar', 'menubar', 'statusbar', 'base_url', 'selector', 'setup' ]);
+  const assertDebugLog = (editor: Editor, expectedLog: unknown) => {
+    const logs: unknown[] = [];
+    const originalLog = console.log;
+    console.log = (arg: unknown) => logs.push(arg);
+    editor.options.debug();
+    console.log = originalLog;
```

Comment:
Please put some sort of try/finally block here, so if `.debug()` throws an exception it doesn't break `console.log` for the entire page

Reply:
We have a few spots of these types of collections maybe we should add a transact type collect function in agar. Then we would be able to add that in a central place.

Reply:
I was expecting an `after` hook to restore the global values, which is exactly what would be good to do with that sort of collect function

Reply:
That would be really nice something like `const { getMessages, reset } = ConsoleHooks.logs({ levels: [ 'error', 'info' ] })` or something.

Reply:
Maybe a training ticket we could make out of that seems simple enough.

Reply:
@TheSpyder  The debug method itself has a try catch encompassing the whole function, so it should be fine as for how it's implemented right now.
I had an after hook in a previous commit, but simplified it later.
Something in agar sounds good to me.